iidm/iidm-api/src/main/java/com/powsybl/iidm/network/BranchAdder.java

```diff
@@ -10,22 +10,23 @@
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public interface BranchAdder<T extends BranchAdder> extends IdentifiableAdder<T> {
+public interface BranchAdder<T extends Branch<? super T>, A extends BranchAdder> extends IdentifiableAdder<T, A> {
```

Comment:
Just for me, why did you write `T extends Branch<? super T>` instead of `T extends Branch<T>`?